apps/challenges/github_utils.py

```diff
@@ -0,0 +1,235 @@
+        content = base64.b64encode(bytes(data, "utf-8")).decode("utf-8")
+        return self.update_content_from_path(path, content)
+
+    def is_repo(self):
+        url = URLS.get("repos").format(self.GITHUB_REPOSITORY)
+        repo_response = self.make_request(url, "GET")
+        return True if repo_response else False
+
+
+@app.task
+def github_challenge_sync(challenge):
+    from .serializers import ZipChallengeSerializer
+
+    for obj in serializers.deserialize("json", challenge):
+        challenge_obj = obj.object
+    serializer = ZipChallengeSerializer(challenge_obj)
+    challenge = serializer.data
+    github = Github_Interface(
+        GITHUB_REPOSITORY=challenge.get("github_repository"),
+        GITHUB_AUTH_TOKEN=challenge.get("github_token"),
+    )
+    if not github.is_repo():
+        return
+    try:
+        # Challenge Non-file field Update
+        non_file_fields = [
+            "title",
+            "short_description",
+            "leaderboard_description",
+            "remote_evaluation",
+            "is_docker_based",
+            "is_static_dataset_code_upload",
+            "start_date",
+            "end_date",
+            "published",
+        ]
+        challenge_config_str = github.get_data_from_path(
+            "challenge_config.yaml"
+        )
+        challenge_config_yaml = yaml.safe_load(challenge_config_str)
+        update_challenge_config = False
+        for field in non_file_fields:
+            # Ignoring commits when no update in field value
+            if (
+                challenge_config_yaml.get(field) is not None
+                and challenge_config_yaml[field] == challenge[field]
+            ):
+                continue
+            update_challenge_config = True
+            challenge_config_yaml[field] = challenge[field]
+        if update_challenge_config:
+            content_str = yaml.dump(challenge_config_yaml, sort_keys=False)
+            github.update_data_from_path("challenge_config.yaml", content_str)
+
+        # Challenge File fields Update
+        challenge_file_fields = [
+            "description",
+            "evaluation_details",
+            "terms_and_conditions",
+            "submission_guidelines",
+        ]
+        for field in challenge_file_fields:
+            if challenge_config_yaml.get(field) is None:
+                continue
+            field_path = challenge_config_yaml[field]
+            field_str = github.get_data_from_path(field_path)
+            if field_str is None or field_str == challenge[field]:
+                continue
+            github.update_data_from_path(field_path, challenge[field])
+    except Exception as e:
+        logger.info("Github Sync unsuccessful due to {}".format(e))
+
+
+@app.task
+def github_challenge_phase_sync(challenge_phase):
+    from .serializers import ChallengePhaseSerializer
+
+    for obj in serializers.deserialize("json", challenge_phase):
+        challenge_phase_obj = obj.object
```

Comment:
create a util method for deserializing and use that

Reply:
Updated.